library/java/org/chromium/net/AndroidNetworkLibrary.java

```diff
@@ -108,8 +109,15 @@ public static void clearTestRootCertificates()
    * Returns true if cleartext traffic to a given host is allowed by the current app.
    */
   public static boolean isCleartextTrafficPermitted(String host) {
-    // TODO(alyssawilk) Implement this method.
-    return true;
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
+      // This API was not implemented before Android N.
+      return true;
+    }
+    // M only supported global checks.
+    if (Build.VERSION.SDK_INT == Build.VERSION_CODES.M) {
```

Comment:
Isn't `M` before `N` - in that case I would expect this API to be not implemented for `M` per your comment from above.

Reply:
in other words - this condition will be never true as we will return early in line 114

Reply:
yes sorry, I'd collapsed https://source.chromium.org/chromium/chromium/src/+/main:net/android/java/src/org/chromium/net/AndroidNetworkLibrary.java;l=357?q=isCleartextTrafficPermitted&ss=chromium into one function and done it incorrectly.  thanks!

Reply:
Thank you for fixing.

If I may suggest something to make it easier to read this code since the flow is still a little bit confusing I think:

```Java
if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
    // This API was not implemented before Android M.
    return true;
} else if (Build.VERSION.SDK_INT == Build.VERSION_CODES.M) {
    // M only supported global checks.
    return NetworkSecurityPolicy.getInstance().isCleartextTrafficPermitted();
} else {
    // The host-specific API was implemented in Android N (which came after Android M).
    return NetworkSecurityPolicy.getInstance().isCleartextTrafficPermitted(host);
}
```

What do you think?